src/wallet/scriptpubkeyman.cpp

```diff
@@ -219,7 +219,7 @@ bool LegacyScriptPubKeyMan::CheckDecryptionKey(const CKeyingMaterial& master_key
         if (keyFail) {
             return false;
         }
-        if (!keyPass && !accept_no_keys && (m_hd_chain.IsNull() || !m_hd_chain.IsNull() && !m_hd_chain.IsCrypted())) {
+        if (!keyPass && !accept_no_keys && (m_hd_chain.IsNull() || (!m_hd_chain.IsNull() && !m_hd_chain.IsCrypted()))) {
```

Comment:
Better to simplify to 
```
if (!keyPass && !accept_no_keys && (m_hd_chain.IsNull() || !m_hd_chain.IsCrypted()))) {
```

Proof. Both functions:
```
bool f(bool a, bool b, bool c) {
    return !a && (b || !b && !c);
}
bool g(bool a, bool b, bool c) {
    return !a && (b || !c);
}
```
are compiled to same binary:
```
        xor     edx, 1
        xor     eax, eax
        or      edx, esi
        test    dil, dil
        cmove   eax, edx
        ret
```